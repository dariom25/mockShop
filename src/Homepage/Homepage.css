.homepage-container {
  display: flex;
  justify-content: center;
}

.homepage-container > a > img {
  border: lightgray solid 3px;
  border-radius: 10px;
  max-width: 100%;
}

.homepage-container > a > img:hover{
    border-color: black;
}