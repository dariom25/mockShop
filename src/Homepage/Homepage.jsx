const Homepage = () => {
  return (
    <div>
      <h1>Welcome to Mocks Online-Schmock Shop</h1>
      <p>Here stands a text which describes why the shop is so special and why you should buy something</p>
      <img src="src/assets/mocksTitlePicture.webp" alt="random image" />
      <a href="shop">Go to the shop</a>
    </div>
  );
};

export default Homepage;
