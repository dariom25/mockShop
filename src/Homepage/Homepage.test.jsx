import { render, screen } from "@testing-library/react";
import Homepage from "./Homepage";
import { it, describe, expect } from "vitest";

describe("Homepage component", () => {
  it("renders correct heading", () => {
    render(<Homepage />);
    expect(screen.getByRole("heading").textContent).toMatch(
      /Welcome to Mocks Online-Schmock Shop/i,
    );
  });
  it("renders homepage according to snapshot", () => {
    const { homepage} = render(<Homepage/>);
    expect(homepage).toMatchSnapshot();
  })
});
