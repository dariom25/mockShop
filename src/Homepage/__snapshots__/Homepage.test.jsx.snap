// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage component > renders homepage according to snapshot 1`] = `undefined`;
